src/app/find-help/accommodation-search/accommodation-search.component.ts

```diff
@@ -1,49 +1,55 @@
 import { Component } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
 import { AccommodationQuery } from './accommodation-search-form/accommodation-search-form.component';
-import { Pagable } from '../../api/pagable';
-import { Location } from '../../../api';
+import { AccommodationsResourceService, OffersAccommodationOffer, AccommodationOffer, Pageable } from '../../../api';
+import { Observable } from 'rxjs';
 
-interface AccommodationOffer {
-  id: number;
-  title: string;
-  description: string;
-  location: Location;
-  guests: number;
-  lengthOfStay: string;
-  hostLanguage: string[];
-}
 @Component({
   selector: 'app-accommodation-search',
   templateUrl: './accommodation-search.component.html',
   styleUrls: ['./accommodation-search.component.scss'],
 })
 export class AccommodationSearchComponent {
   results: AccommodationOffer[] = [];
-  total: number = NaN;
+  total?: number = undefined;
   loading = false;
-  constructor(private http: HttpClient) {}
+  constructor(private accommodationsResourceService: AccommodationsResourceService) {}
 
-  search(query: AccommodationQuery) {
+  search(searchCriteria: AccommodationQuery) {
     this.loading = true;
-    this.http
-      .get<Pagable<AccommodationOffer>>(
-        `/api/accommodations/` + (query.location ? `${query.location.region}/${query.location.city}` : ''),
-        {
-          params: query.guests ? { capacity: query.guests } : {},
-        }
-      )
-      .subscribe({
-        next: (results) => {
-          this.results = results.content;
-          this.total = results.totalElements;
-          this.loading = false;
-        },
-        error: () => {
-          this.results = [];
-          this.total = NaN;
-          this.loading = false;
-        },
-      });
+
+    // TODO specify the params; at least sorting is configurable from the UI
+    const pageRequest: Pageable = {
+      // page?: number;
+      // size?: number;
+      // sort?: Array<string>;
+    };
+
+    let resultsObservable: Observable<OffersAccommodationOffer>;
```

Comment:
Create a method that return the `resultsObservable` so that we don't have a `let` variable. 
(`let` variables make it harder to understand the code, as it's not obvious on first glance were all the possible initialisers are. At least for me)

Reply:
Done. This also allowed me to not declare so many types explicitly which is nice.